packages/pages/src/dev/server/ssr/generateTestData.ts

```diff
@@ -159,18 +180,30 @@ export const generateTestDataForPage = async (
       resolve(parsedData);
     });
   });
-};
+}
+
+const getCommonArgs = (
+  featuresConfig: FeaturesConfig,
+  projectStructure: ProjectStructure
+) => {
+  const args = ["pages", "generate-test-data", "--printDocuments"];
+
+  args.push("--featuresConfig", prepareJsonForCmd(featuresConfig));
+
+  const sitesConfigPath =
```

Comment:
Can this be done elsewhere so `siteStream` can be passed in instead of `projectStructure`, which is only used to get `siteStream`.